server/src/test/java/io/crate/planner/operators/LogicalPlanTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner.operators;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import io.crate.analyze.WhereClause;
+import io.crate.analyze.relations.AbstractTableRelation;
+import io.crate.analyze.relations.AliasedAnalyzedRelation;
+import io.crate.analyze.relations.DocTableRelation;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.Reference;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.metadata.table.Operation;
+import io.crate.planner.node.dql.join.JoinType;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SQLExecutor;
+import io.crate.types.DataTypes;
+
+public class LogicalPlanTest extends CrateDummyClusterServiceUnitTest {
```

Comment:
There is already a `LogicalPlannerTest`, would it be possible to add these cases there, or use a more specific name? There is also already a `JoinTest` and a `UnionPlannerTest`. Maybe they fit?

Otherwise I think it may not be clear what the difference between `LogicalPlanTest` and `LogicalPlannerTest` is and when to use/extend them.

Reply:
The main reason for putting everything into one class was to not duplicate all the test data creation over multiple tests. However, I am ok splitting it.

Reply:
> This sounds like it fixes a bug? Can a user already craft a query that triggers an issue? If so - this would need a release notes entry and a backport.

AFAIK this can currently not be triggered by a user query. The hash-join internally handles the wrong relation names correctly. However, it does cause problems once i put the MoveConstantJoinConditionsBeneathNestedLoop in place together with another issue. I wanted to have seperate pr's to fix that. But maybe it is easier if just put multiple commits into a one pr to get the full picture.

Reply:
>  But maybe it is easier if I just put multiple commits into a single pr to get the full picture.

Fine to keep it separate, just wanted to make sure.

> The main reason for putting everything into one class was to not duplicate all the test data creation over multiple tests. However, I am ok splitting it.

Also okay to keep it in a single file, but then I think it needs a more descriptive name that is not as ambiguous with other existing test classes

Reply:
> Fine to keep it separate, just wanted to make sure.

I spent more time on testing and i believe this is not a user-facing issue since 'getRelationNames()' is only internally used in some optimizer rules.

Reply:
I would like to use the method later in an optimizer rule, and it would good if it then works correctly ;-)